rclcpp/include/rclcpp/macros.hpp

```diff
@@ -23,14 +23,33 @@
   Class(const Class &) = delete; \
   Class & operator=(const Class &) = delete;
 
-/* Defines a make_shared static function on the class using perfect forwarding.
+/* Defines typedefs and static functions for using the Class with smart pointers.
  *
  * Use in the public section of the class.
  * Make sure to include <memory> in the header when using this.
  */
-#define RCLCPP_MAKE_SHARED_DEFINITIONS(Class) \
-  typedef std::shared_ptr<Class> SharedPtr; \
- \
+#define RCLCPP_SMART_PTR_DEFINITIONS(Class) \
+  RCLCPP_SHARED_PTR_DEFINITIONS(Class) \
+  RCLCPP_WEAK_PTR_DEFINITIONS(Class) \
+  RCLCPP_UNIQUE_PTR_DEFINITIONS(Class)
+
+/* Defines typedefs and static functions for using the Class with smart pointers.
+ *
+ * Same as RCLCPP_SMART_PTR_DEFINITIONS expect it excludes the static
+ * Class::make_unique() method definition which does not work on classes which
+ * are not CopyConstructable.
+ *
+ * Use in the public section of the class.
+ * Make sure to include <memory> in the header when using this.
+ */
+#define RCLCPP_SMART_PTR_DEFINITIONS_NOT_COPYABLE(Class) \
+  RCLCPP_SHARED_PTR_DEFINITIONS(Class) \
+  RCLCPP_WEAK_PTR_DEFINITIONS(Class) \
+  __RCLCPP_UNIQUE_PTR_TYPEDEF(Class)
+
+#define __RCLCPP_SHARED_PTR_TYPEDEF(Class) typedef std::shared_ptr<Class> SharedPtr;
```

Comment:
Just an aesthetic suggestion. Given that we're redoing the macros, we can start using `using` instead of `typedef`:

```c++
#define __RCLCPP_SHARED_PTR_TYPEDEF(Class) using SharedPtr = std::shared_ptr<Class>;
```

this would make the code more uniform if any of the aliases require a template, since `typedef` can't be templated.